### ISSUE TITLE
HttpClient needs to accept options

### ISSUE THREAD

**Issue by peledies** [AuthorizeNet/sdk-php#276]
`sdk-php/lib/net/authorize/util/HttpClient.php` is not configurable from what I can tell.

it would be nice to be able to add `curl` options to the request, specifically to specify the ssl version.
```php
curl_setopt($curl_request, CURLOPT_SSL_VERIFYHOST, 2);
```

Similar to the way the Stripe SDK works, Example:
```php
\Stripe\ApiRequestor::setHttpClient(
    new \Stripe\HttpClient\CurlClient(
        [ CURLOPT_SSLVERSION => CURL_SSLVERSION_TLSv1_2 ]
    )
);
```

**Comment by britbarn** [comment]
Agreed that this would be nice to have. Updating linux might not be able to happen in some cases.

**Comment by ashtru** [comment]
Thanks @peledies, this looks like a good feature to have. 
We can have a look into adding this configuration in the SDK.
